### ISSUE TITLE
[DEV]  the default partition disk detection logic in pre.rh.rhels7 is out-of-date, since "/lib/udev/edd_id" is not shipped in redhat7.2 anymore

### ISSUE THREAD

**Issue by immarvin** [xcat2/xcat-core#371]
The following code slice in "pre.rh.rhels7" is used to detect the default disk to partition, however, "/lib/udev/edd_id " is not shipped in redhat7 anymore

````
-bash-4.2# find . /lib/udev/
.
./.fbtermrc
./.bash_profile
/lib/udev/
/lib/udev/mtd_probe
/lib/udev/rules.d
/lib/udev/rules.d/60-persistent-storage-tape.rules
/lib/udev/rules.d/60-alias-kmsg.rules
/lib/udev/rules.d/98-rdma.rules
/lib/udev/rules.d/60-keyboard.rules
/lib/udev/rules.d/81-kvm-ibm.rules
/lib/udev/rules.d/60-persistent-serial.rules
/lib/udev/rules.d/75-net-description.rules
/lib/udev/rules.d/69-dm-lvm-metad.rules
/lib/udev/rules.d/42-usb-hid-pm.rules
/lib/udev/rules.d/62-multipath.rules
/lib/udev/rules.d/91-drm-modeset.rules
/lib/udev/rules.d/60-persistent-input.rules
/lib/udev/rules.d/60-net.rules
/lib/udev/rules.d/95-udev-late.rules
/lib/udev/rules.d/80-drivers.rules
/lib/udev/rules.d/50-udev-default.rules
/lib/udev/rules.d/11-dm-lvm.rules
/lib/udev/rules.d/70-power-switch.rules
/lib/udev/rules.d/40-redhat.rules
/lib/udev/rules.d/11-dm-mpath.rules
/lib/udev/rules.d/80-net-name-slot.rules
/lib/udev/rules.d/73-idrac.rules
/lib/udev/rules.d/64-btrfs.rules
/lib/udev/rules.d/95-dm-notify.rules
/lib/udev/rules.d/60-persistent-storage.rules
/lib/udev/rules.d/60-raw.rules
/lib/udev/rules.d/61-accelerometer.rules
/lib/udev/rules.d/60-cdrom_id.rules
/lib/udev/rules.d/71-seat.rules
/lib/udev/rules.d/63-md-raid-arrays.rules
/lib/udev/rules.d/90-vconsole.rules
/lib/udev/rules.d/75-tty-description.rules
/lib/udev/rules.d/60-persistent-v4l.rules
/lib/udev/rules.d/65-md-incremental.rules
/lib/udev/rules.d/99-systemd.rules
/lib/udev/rules.d/10-dm.rules
/lib/udev/rules.d/75-probe_mtd.rules
/lib/udev/rules.d/13-dm-disk.rules
/lib/udev/rules.d/73-seat-late.rules
/lib/udev/rules.d/60-persistent-alsa.rules
/lib/udev/rules.d/78-sound-card.rules
/lib/udev/rules.d/70-uaccess.rules
/lib/udev/cdrom_id
/lib/udev/scsi_id
/lib/udev/accelerometer
/lib/udev/udev-kvm-check
/lib/udev/rename_device
/lib/udev/collect
/lib/udev/ata_id
/lib/udev/hwdb.d
/lib/udev/hwdb.d/22-bluetooth-vendor-product.hwdb
/lib/udev/hwdb.d/22-OUI.hwdb
/lib/udev/hwdb.d/22-pci-classes.hwdb
/lib/udev/hwdb.d/20-usb-vendor-model.hwdb
/lib/udev/hwdb.d/20-OUI.hwdb
/lib/udev/hwdb.d/20-acpi-vendor.hwdb
/lib/udev/hwdb.d/22-usb-vendor-model.hwdb
/lib/udev/hwdb.d/62-keyboard.hwdb
/lib/udev/hwdb.d/20-usb-classes.hwdb
/lib/udev/hwdb.d/60-keyboard.hwdb
/lib/udev/hwdb.d/20-pci-vendor-model.hwdb
/lib/udev/hwdb.d/20-bluetooth-vendor-product.hwdb
/lib/udev/hwdb.d/22-pci-vendor-model.hwdb
/lib/udev/hwdb.d/22-usb-classes.hwdb
/lib/udev/hwdb.d/20-pci-classes.hwdb
/lib/udev/v4l_id
````

````
#time to ascertain fstype and PReP/UEFI/legacy
#also, find first available block device (sda or vda likely)
#TODO: pick a likely non-SAN target if possible
144 shopt -s nullglob
145 for disk in /dev/vd*[^0-9];do
146         if [ -z "$firstdirectdisk" ]; then firstdirectdisk=$disk; fi #remember first disk as a guess of medi    um resort
147         eddname=$(/lib/udev/edd_id $disk 2> /dev/null)
148         if [ ! -z "$eddname" -a "$eddname" = "int13_dev80" ]; then
149                 instdisk=$disk
150                 break
151         fi
152 done
153 
154 if [ -z "$instdisk" ]; then
155         for disk in /dev/sd*[^0-9]; do
156                 eddname=$(/lib/udev/edd_id $disk 2> /dev/null)
157                 if [ ! -z "$eddname" -a "$eddname" = "int13_dev80" ]; then
158                         instdisk=$disk
159                         break
160                 fi
161                 currdriver=`udevadm info --attribute-walk --name $disk |grep DRIVERS|grep -v '""'|grep -v '"    sd"'|head -n 1|sed -e 's/[^"]*"//' -e 's/"//'`
162                 case "$currdriver" in
163                 "ata_piix4"|"PMC MaxRAID"|"ahci"|"megaraid_sas") #certainly direct
164                         if [ -z "$firstdirectdisk" ]; then firstdirectdisk=$disk; fi #remember first disk as     a guess of medium resort
165                         ;;
166                 "mptsas"|"mpt2sas"|"mpt3sas") #*PROBABLY* not SAN, but SAS SAN is possible
167                         if [ -z "$probablyfirstdirectdisk" ]; then probablyfirstdirectdisk=$disk; fi #rememb    er first disk as a guess of medium resort
168                         ;;
169                 *)
170                         if [ -z "$firstdisk" ]; then firstdisk=$disk; fi #remember first disk as a guess of     medium resort
171                         ;;
172                 esac
173         done
174 fi
175 
176 if [ -z "$instdisk" ]; then
177         if [ ! -z "$firstdirectdisk" ]; then
178                 instdisk=$firstdirectdisk
179         elif [ ! -z "$probablyfirstdirectdisk" ]; then
180                 instdisk=$probablyfirstdirectdisk
181         elif [ ! -z "$firstdisk" ]; then
182                 instdisk=$firstdisk
183         fi
184 fi
185 # Could not determine the install disk for whatever reason,
186 # try the default /dev/sda
187 if [ -z "$instdisk" ]; then
188     instdisk="/dev/sda"
189 fi
````

**Comment by whowutwut** [comment]
@immarvin , Is this something we should fix in the RC1 build for 2.11?  @tingtli FYI

**Comment by immarvin** [comment]
hi @whowutwut , I think this issue should not be fixed in 2.11 since it involves a lot of code change and will introduce great influence. 

In 2.11, I just added the logic to determine the partition disk for powerkvm https://github.com/xcat2/xcat-core/pull/456.

I think the logic to determine the partition disk needs to be refined and made consistent across linux distributions , I opened a ticket to trace it https://github.com/xcat2/xcat-core/issues/458.
